### PR TITLE
MHV-50222

### PR DESCRIPTION
**Pull request by gerard-sullivan**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

medications URLs were changed for this browser back button expection
-     path={['/', '/:page']} 
-     path="/prescription/:prescriptionId"
added a state value for saving the prescription id that user clicked lastly that will be used scroll the exact location when user click the browser back button.

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-50222

## Testing done

Tested scenario:  # I is on the 6th of their med list, and I scroll down to the last medication of the medications list page.
1. I clicks on that medication to view its details.
2. After viewing the details, I want to go back to the list page using the browser’s back button.
3. With the proposed enhancement, I land back on the 6th page of the medications list page with the last item on that page still in view, exactly where I left off, rather than starting back at the top of the 1th page.

## Screenshots
NO UI changes
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ x No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by KolbyVA — CHANGES_REQUESTED** [review]
Is it possible to write/update some unit tests for these changes?

File: src/applications/mhv/medications/reducers/prescriptions.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ const initialState = {
    * Pagination received form meta object in prescriptionsList payload
    */
   prescriptionsPagination: undefined,
+  /**
+   * Selected PrescriptionId that is used for scroll to exact location when clicked browser back button in details page
+   */
+  prescriptionId: undefined,
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/reducers/prescriptions.js:20]: Wouldn't it persist within `prescriptionDetails`?

**Review by vmenshutin-bylight — CHANGES_REQUESTED** [review]

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js [hunk 2]
```diff
@@ -82,6 +82,23 @@ class MedicationsDetailsPage {
...
+        prescriptionDetails.data.attributes.prescriptionId
+      }`,
+      prescriptionDetails,
+    ).as('prescriptionDetails');
+    cy.get(
+      `#card-header-${
+        prescriptionDetails.data.attributes.prescriptionId
+      } > [data-testid="medications-history-details-link"]`,
+    )
+      .should('be.visible')
```
Comment (Asterix22) [inline src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js:98]: split out should('be.visible') from .click();  checking for visibility before clicking is probably more stable.

**Review by Asterix22 — CHANGES_REQUESTED** [review]

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js (hunk 2 above)
Comment (RajiVenkatesh18) [reply src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js]: yes, fixed that one and ran 150x . Thanks!

**Review by vmenshutin-bylight — APPROVED** [review]

**Review by Asterix22 — APPROVED** [review]

**Review by KolbyVA — APPROVED** [review]

**Review by ataker — APPROVED** [review]

**Pull request merged**